validator/client/beacon-api/BUILD.bazel

```diff
@@ -1,15 +1,46 @@
 load("@prysm//tools/go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
 # Gazelle will flag this for removal since it's only being used with the use_beacon_api flag, but it should be kept
 # keep
+# gazelle:build_tag use_beacon_api
 go_library(
     name = "go_default_library",
-    srcs = ["beacon_api_validator_client.go"],
+    srcs = [
+        "beacon_api_validator_client.go",
+        "genesis.go",
+    ],
     importpath = "github.com/prysmaticlabs/prysm/v3/validator/client/beacon-api",
     visibility = ["//validator:__subpackages__"],
     deps = [
+        "//api/gateway/apimiddleware:go_default_library",
+        "//beacon-chain/rpc/apimiddleware:go_default_library",
         "//proto/prysm/v1alpha1:go_default_library",
         "//validator/client/iface:go_default_library",
+        "@com_github_ethereum_go_ethereum//common/hexutil:go_default_library",
+        "@com_github_pkg_errors//:go_default_library",
         "@io_bazel_rules_go//proto/wkt:empty_go_proto",
     ],
 )
+
+# Gazelle will flag this for removal since it's only being used with the use_beacon_api flag, but it should be kept
+# keep
+# gazelle:build_tag use_beacon_api
```

Comment:
```suggestion
# gazelle:build_tag use_beacon_api
```

You shouldn't need the keep directive if the build_tag directive is working properly.